yarn-project/sequencer-client/src/sequencer/sequencer.ts

```diff
@@ -540,18 +551,59 @@ export class Sequencer {
     return orderAttestations(attestations, committee);
   }
 
+  protected async createProofClaimForPreviousEpoch(slotNumber: bigint): Promise<EpochProofQuote | undefined> {
+    // Find out which epoch we are currently in
+    const epochForBlock = await this.publisher.getEpochForSlotNumber(slotNumber);
+    if (epochForBlock < 1n) {
+      // It's the 0th epoch, nothing to be proven yet
+      this.log.verbose(`First epoch has no claim`);
+      return undefined;
+    }
+    const epochToProve = epochForBlock - 1n;
```

Comment:
As discussed, this won't work if the previous epoch was empty. Suppose proven tip is at the end of epoch `n-1`, and no blocks land in `n`, then the epoch to be proven is `n+1`.

I suggest just fixing in https://github.com/AztecProtocol/aztec-packages/issues/8855